### DISCUSSION TITLE
Mac OS X 10.11.6 installed 6.0.3, updated the package to 6.1.2, stopped running tsh, teleport, ...

### DISCUSSION THREAD

**Discussion by joyshmitz** [gravitational/teleport#6631] [Q&A] [answered]
The question is, how do they update correctly on a Mac?

I guess these are compatibility issues 10.11 and 10.12 in specific Symbol called `_clock_gettime`, which did not exists in OS X <10.12
maybe you stopped supporting 10.11, unfortunately I did not find in the release notes that 10.11 is not supported

`macpro% tsh
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /usr/local/bin/tsh
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _clock_gettime
  Referenced from: /usr/local/bin/tsh
  Expected in: /usr/lib/libSystem.B.dylib
...`

**Comment by webvictim** [comment, marked as answer]
Yes, this looks like the minimum version of MacOS supported by Golang changed at some point.

Here is a table comparing `teleport`/`tsh` version to Golang version:

| Teleport version | Golang version used |
| - | - |
| `7.x` | 1.16.2 |
| `6.x` | 1.15.5 |
| `5.x` | 1.15.5 |
| `4.4.x` | 1.14.4 |
| `4.3.x` | 1.13.2 |

If you find that newer versions of `tsh` aren't working, you can try using one of the older ones.

I will see if we can get some documentation added relating to minimum MacOS version supported.

**Reply by joyshmitz** [reply]
Thank you so much for all your hard work. Great product.
